atlasdb-client/src/main/java/com/palantir/atlasdb/logging/LoggingArgs.java

```diff
@@ -58,6 +59,7 @@
     }
 
     private static volatile KeyValueServiceLogArbitrator logArbitrator = KeyValueServiceLogArbitrator.ALL_UNSAFE;
+    private static volatile Optional<Boolean> allSafeForLogging = Optional.empty();
```

Comment:
If I'm not mistaken `allSafeForLogging` is only written here by the static initializer and thereafter read to and written from in `setAllSafeForLogging`, so all accesses across threads should already have the happens-before relationship.